time/time.c

```diff
@@ -181,8 +183,11 @@ struct tm *gmtime_r(const time_t *timep, struct tm *res)
 
 	res->tm_wday = (day + 4) % 7;
 
-	res->tm_year = 70 + day / 365;
-	leapcnt = leapcount(res->tm_year + 1900);
+	// Limit the following calculations to a 400 year period
+	full_400_years = day / DAYS_IN_400_YEARS;
+	day = day % DAYS_IN_400_YEARS;
+
+	leapcnt = leapcount(1970 + day / 365);  // This leap day count is approximate, but works within a 400 year period.
```

Comment:
Perhaps it would be better to fix `leapcount()` instead

Reply:
`leapcount()` is no longer used in `gmtime_r()` after my fixes, but it's still used in `_mktimeSkel()`. I can try to improve that function too.